code/modules/mob/mob_movement.dm

```diff
@@ -81,6 +81,16 @@
 
 	if(!mob.Process_Spacemove(direct))
 		return FALSE
+
+
+	var/delay = mob.movement_delay() // Yoggybear start
+
+	if(Can_ShadowWalk(mob))
+		if(Process_ShadowWalk(direct))

```

Comment:
I really don't want this ``/client/Move()`` when this is only used in 1 gamemode. Can't you implement this using components? If not, look at ``/client/proc/Process_Incorpmove()``.

Reply:
I'm gonna give my hand at making it a component, shouldn't be too hard but it'll take me some time